### ISSUE TITLE
`wp_get_global_styles`: return values instead of CSS Custom Properties

### ISSUE THREAD

**Issue by oandregal** [WordPress/gutenberg#49712]
Related https://github.com/WordPress/gutenberg/issues/49693 https://github.com/WordPress/gutenberg/issues/45171

## What problem does this address?

The `wp_get_global_styles` function returns the content in the `theme.json`.

For example, for a content such as:

```json
"core/post-terms": {
    "typography": { "fontSize": "var(--wp--preset--font-size--small)" }
}
```

The expected result is:

```php
(
    [typography] => Array( [fontSize] => var(--wp--preset--font-size--small) )
)
```

Some environments don't have support for CSS Custom Properties. For example, mobile or email templates. They have to parse the content to understand it and convert it to a value they can use.

## What is your proposed solution?

There are two alternatives we could explore to help them use `theme.json` provided values:

- Add some new parameter to `wp_get_global_styles` that tells the function to resolve the CSS Custom Properties to the value they reference.
- Create a new function, for example `wp_get_global_styles_values`, that does the same.

**Comment by samnajian** [comment]
@oandregal I'd rather go with your second proposal since:

1) The use case is pretty limited for now, so it's better to not adjust the main, more multi-purpose function to do even more
2) `wp_get_global_styles` is already receiving 2 parameters in the form of arrays, which in the absence of proper typing on the language level are confusing enough, so refraining from adding a new parameter to the same rather complex input parameters would keep the complexity the same and not add to it.

So basically by introducing a new function, we can keep the complexity to the underlying classes and not push it to the user side.

**Comment by oandregal** [comment]
Relevant thread for this conversation, so we can see everything together https://github.com/WordPress/gutenberg/pull/50484/files#r1192502874

**Comment by oandregal** [comment]
This has been fixed by https://github.com/WordPress/gutenberg/pull/50484
